make/modules/java.base/Java.gmk

```diff
@@ -37,7 +37,8 @@ EXCLUDE_FILES += \
 
 EXCLUDES += java/lang/doc-files \
     java/lang/classfile/snippet-files \
-    java/lang/classfile/components/snippet-files
+    java/lang/classfile/components/snippet-files \
+    jdk/lang/foreign/snippet-files
```

Comment:
I can't find this directory in the source tree, but there is a `src/java.base/share/classes/java/lang/foreign/snippet-files`. Is that what you want to exclude?

Reply:
If possible, we should simply exclude all files in directories that have `-` (minus sign) in their name; this is the intentional design to prevent javac from compiling those classes as package names cannot include `-`.

Reply:
I agree that this piecemeal approach is not good. I think there is a JBS enhancement request to filter all "snippet-files" and "javadoc-files" everywhere. But maybe we can make it broader? Filtering on just `-` makes me a bit nervous though; it seems like it could unintentionally break at some point. But maybe filter out all `-files`?

Reply:
It would be good if snippets were compiled (so that the syntax and correctness is ensured) but not included in shipped artifacts.